ansible/roles/open-env-azure-remove-user-from-subscription/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+- name: Log into Azure CLI
+  command: >
+    az login --service-principal -u {{ azure_service_principal }}
+    -p {{ azure_password }} --tenant {{ azure_tenant }}
+
+- name: Retrieve which Pool ID is allocated for this user from the database
+  command: >
+    curl {{ az_function_show }}{{ project_tag }}
+  register: poolid
+
+- name: Write out the assigned Pool ID
+  debug:
+    msg: "{{ poolid.stdout }}"
+
+- name: Get facts for the subscription by name
+  azure.azcollection.azure_rm_subscription_info:
+    auth_source: cli
+    name: "{{ poolid.stdout }}"
+  register: assignedsubscription
+
+- name: Write out assignedsubscription
+  debug:
+    msg: "{{ assignedsubscription }}"
+
+- name: Get subscription ID
+  set_fact:
+    subscription_id: "{{ assignedsubscription['subscriptions'][0].subscription_id }}"
+
+- name: Set Default Subscription in Azure CLI
+  command: >
+    az account set --subscription "{{ subscription_id }}"
+
+- name: Get all resource groups in the subscription
+  azure.azcollection.azure_rm_resourcegroup_info:
+    auth_source: cli
+    subscription_id: "{{ subscription_id }}"
+    tenant: "{{ azure_tenant }}"
+  register: allresourcegroups
+
+- name: Delete all resource groups owned by the subscription (RG's created by the lab user)
+  when: allresourcegroups['resourcegroups']|length>0
+  azure.azcollection.azure_rm_resourcegroup:
+    auth_source: cli
+    subscription_id: "{{ subscription_id }}"
+    name: "{{ item.name }}"
+    location: "{{ item.location }}"
+    force_delete_nonempty: yes
+    state: absent
+  loop: "{{ allresourcegroups['resourcegroups'] }}"
+
+- name: Get subscription FQID
+  set_fact:
+    subscriptionfqid: "{{ assignedsubscription['subscriptions'][0].fqid }}"
+
+- name: Clean up all Tags on the Subscription 
+  command: >
+    az tag delete --resource-id {{ subscriptionfqid }} --yes
+
+- name: Get the user's object from Active Directory
+  azure.azcollection.azure_rm_aduser_info:
+    auth_source: cli
+    user_principal_name: "{{ email }}"
+    tenant: "{{ azure_tenant }}"
+  register: azuser
+
+- name: Get CONTRIBUTOR Role Definition
+  azure.azcollection.azure_rm_roledefinition_info:
+    auth_source: cli
+    scope: "{{ subscriptionfqid }}"
+    role_name: Contributor
+  register: contributor_role_definition
+
+- name: REMOVE the user's CONTRIBUTOR role from the subscription
+  azure.azcollection.azure_rm_roleassignment:
+    auth_source: cli
+    scope: "{{ subscriptionfqid }}"
+    assignee_object_id: "{{ azuser['ad_users'][0].object_id }}"
+    role_definition_id:
+      "{{ contributor_role_definition['roledefinitions'][0].id }}"
```

Comment:
Why is this string on the next line? I think if we do put a string on a subsequent line we should use a yaml block quote marker like `>-`. I know this is valid, but it feels weird.

Reply:
fixed